Classes/Private/Manager/Video/OCTVideoEngine.m

```diff
@@ -100,6 +100,46 @@ - (BOOL)setupWithError:(NSError **)error
     return YES;
 }
 
+- (BOOL)switchToCameraFront:(BOOL)front error:(NSError **)error
+{
+    DDLogVerbose(@"%@: switchToCameraFront %d", self, front);
+
+    AVCaptureDevicePosition position = front ? AVCaptureDevicePositionFront : AVCaptureDevicePositionBack;
+
+    NSArray *inputs = [self.captureSession inputs];
+
+    AVCaptureInput *current = [inputs firstObject];
+    if ([current isKindOfClass:[AVCaptureDeviceInput class]]) {
+        AVCaptureDeviceInput *inputDevice = (AVCaptureDeviceInput *)current;
+        if (inputDevice.device.position == position) {
+            return YES;
+        }
+    }
+
+    for (AVCaptureInput *input in inputs) {
+        [self.captureSession removeInput:input];
+    }
+
+    AVCaptureDevice *camera = [self getDeviceForPosition:position];
+
+    AVCaptureDeviceInput *videoInput = [AVCaptureDeviceInput deviceInputWithDevice:camera error:error];
+
+    if (! videoInput) {
+        return NO;
+    }
+
+    if (! [self.captureSession canAddInput:videoInput]) {
+        // fill error?
```

Comment:
I think it is okay to ignore it as there should be no error in all normal cases.